test/e2e/utils.go

```diff
@@ -0,0 +1,225 @@
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// lookForStringInPodExec looks for expectedString in the output of command
+// executed in the specified pod container every 2 seconds until the timeout
+// is reached or the string is found. Returns an error if the string was not found.
+func lookForStringInPodExec(ns, pod, container string, command []string, expectedString string, timeout time.Duration) error {
+	cmdPath, err := exec.LookPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"exec", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns), "--"}
+	args = append(args, command...)
+	if err := lookForString(cmdPath, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookForStringInPodLog looks for the given string in the log of the
+// specified pod container every 2 seconds until the timeout is reached
+// or the string is found. Returns an error if the string was not found.
+func lookForStringInPodLog(ns, pod, container, expectedString string, timeout time.Duration) error {
+	cmdPath, err := exec.LookPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"logs", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns)}
+	if err := lookForString(cmdPath, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookForString looks for the given string using cmd and args every
+// 2 seconds until the timeout is reached or the string is found.
+// Returns an error if the string was not found.
+func lookForString(cmd string, args []string, expectedString string, timeout time.Duration) error {
+	for t := time.Now(); time.Since(t) < timeout; time.Sleep(2 * time.Second) {
```

Comment:
Any reason not to use `wait.PollImmediate` here?  Sorry I didn't think of it sooner.  We can leave it for a follow-up.

Reply:
I've moved the logic to `wait.PollImmediate`.